### DISCUSSION TITLE
什么时候升级.net8啊?加上打开文件和文件夹

### DISCUSSION THREAD

**Discussion by xingzuhui** [HandyOrg/HandyControl#1506] [Ideas]
什么时候升级.net8啊,因为8加了重写的打开文件和打开文件夹的原生服务.
另外,是不是没有包含所有的控件重写,好像还有很多控件的. 有些是无外观的element,也不需要重写

**Comment by NaBian** [comment]
8 升了，其它还有什么控件？

**Reply by xingzuhui** [reply]
还有控件很多,包括:
1.现有控件的样式扩展,比如button的icon的上下左右,其实都需要,就算不写大量的扩展,也应该guide里把怎么自己写,写清楚.我的意思是,在哪个层级basestyle还是basebasestyle写有什么效果写清楚.
2.还有好多控件不存在呢,不说写一些不断更新数据,如股票价格,需要缓存的控件,也不说需要控制动画,比方风扇转速和水管气管流速的控件,就是普通的flentui的控件也不全.winui3和wpfui.  我有空就专开一贴,做看到的新控件.
打开文件和文件夹的控件,确实很常用.
做ui的目的是最小代价的复用, 这是我最近的体会. 程序的原始控件就可以做所有的事情,那么怎么把另外人的想法,最快的传达出来,控件就是细粒度,通用的一种方法. 更高的是usercontrol,template,desigmpattern. app

**Reply by xingzuhui** [reply]
我增加了如下2种样式,还要加button的样式.
![image](https://github.com/user-attachments/assets/68d29f21-2b8e-4d71-8eeb-7dbd2ab8814c)
![image](https://github.com/user-attachments/assets/2bf23655-1a3b-45fd-9c8b-a4e3c369ad3f)
这些都是程序常用,一个是表示状态的红绿灯,一个是显示不可修改项,比如不可改属性,比如订单号,只是系统生成,给你看的.

**Reply by xingzuhui** [reply]
1.演示系统太老的架构了,虽然是效率因为做的缓存和lazy,很好,但是跟现在host控制全场格格不入.要现代化.
2.改的地方,我知道的,还要命令代替事件,communitytookit代替lighttookit,还有语言适合C#12和net8,新特性都不用,学习的人不是只用ui.所有的东西都模仿,你如果不说,看演示代码的也不知道什么是最新的用发法. 特别厉害的人,不需要模仿的人,又看不上我们了.  我看iNKORE.UI.WPF.Modern.Gallery 用的是host的演示程序,虽然他没lasy示例,
3.全面ioc改进,用微软的或者dryioc. 除了static的工具类,(还包括entry不用)自己管理生命周期,其他所有都需要.
4.总之需要新分支,不是编译平台分支.是带新语言,dotnet特性分支.

**Reply by xingzuhui** [reply]
我遇到好多需要加标题的,怎么办呢?原先我是继承老控件的defaultstyle.  但是现在好多控件都有这个需求,比如 当前登录用户:admin  这样的控件,明显就是需要splitbutton带titile,但是复杂控件显示title,就要重写所有模板了.那工作量很大,如果是好多控件都要这样做,不划算啊.   怎么扩展呢?如果不能扩展,应该把所有的附加属性都显示了把.
其实,扩展要在相对的basestyle上做,子具体style继承,但是就要重写style,后面升级怎么办呢?
